### ISSUE TITLE
b2: Stat: unexpected end of JSON input

### ISSUE THREAD

**Issue by sumnerboy12** [restic/restic#1377]
## Output of `restic version`
```
restic 0.7.3
compiled with go1.9 on freebsd/amd64
```

## How did you run restic exactly?
Run via cron script which runs;
```
source /etc/repo.config
restic forget --prune --keep-daily 7 --keep-monthly 12 --keep-yearly 5
```
where `repo.config` contains my B2 account key/id and restic repository name/password.

## What backend/server/service did you use?
Backblaze B2

## Expected behavior
Successful forget and prune

## Actual behavior
Get the following output...
```
snapshots for (host [nas.home], paths [/mnt/pool01/personal]):

keep 9 snapshots:
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
fe2ab887  2017-10-18 00:12:28  nas.home                /mnt/pool01/personal
28649bcd  2017-10-17 00:10:44  nas.home                /mnt/pool01/personal
f3084fe4  2017-10-16 08:56:51  nas.home                /mnt/pool01/personal
398ea1f1  2017-10-15 00:10:23  nas.home                /mnt/pool01/personal
cdb97ac8  2017-10-14 00:10:40  nas.home                /mnt/pool01/personal
4846b8bc  2017-10-13 00:10:18  nas.home                /mnt/pool01/personal
4896b809  2017-10-11 00:00:45  nas.home                /mnt/pool01/personal
ca6898e6  2017-09-30 00:12:08  nas.home                /mnt/pool01/personal
a4d586a3  2017-08-31 01:00:44  nas.home                /mnt/pool01/personal

remove 6 snapshots:
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
ad4f9cb3  2017-10-10 00:00:48  nas.home                /mnt/pool01/personal
b860bfc6  2017-10-09 00:12:07  nas.home                /mnt/pool01/personal
8561543a  2017-10-08 00:16:09  nas.home                /mnt/pool01/personal
92f43740  2017-10-07 00:13:10  nas.home                /mnt/pool01/personal
d8178b87  2017-10-06 00:11:30  nas.home                /mnt/pool01/personal
b3e8aba5  2017-10-05 00:12:17  nas.home                /mnt/pool01/personal

6 snapshots have been removed, running prune
counting files in repo
building new index for repo
pack file cannot be listed 3d3de1c9: Stat: unexpected end of JSON input
pack file cannot be listed 515669c4: Stat: unexpected end of JSON input
pack file cannot be listed 6fe1bb09: Stat: unexpected end of JSON input
pack file cannot be listed 7ab5f434: Stat: unexpected end of JSON input
pack file cannot be listed 7ab5fdb7: Stat: unexpected end of JSON input
pack file cannot be listed 7aafe7a1: Stat: unexpected end of JSON input
pack file cannot be listed de71db86: Stat: unexpected end of JSON input
[1:59:29] 100.00%  15949 / 15949 packs

repository contains 15942 packs (71328 blobs) with 75.117 GiB bytes
processed 71328 blobs: 0 duplicate blobs, 0B duplicate
load all snapshots
find data that is still in use for 9 snapshots
[5:41] 100.00%  9 / 9 snapshots

number of used blobs is larger than number of available blobs!
Please report this error (along with the output of the 'prune' run) at
https://github.com/restic/restic/issues/new
```

## Steps to reproduce the behavior
Haven't reproduced (or tried to). The prune takes many many hours and have just received this error, hence reporting (due to the request in the error log).

## Do you have any idea what may have caused this?
None. Have noticed most of my B2 backups (4 out of 5) have been reporting index errors when checking or pruning.

## Do you have an idea how to solve the issue?
None.

**Comment by fd0** [comment]
Thanks for the report! This is caused by a bug in the b2 backend described in #1351, it has been fixed a couple of days ago in #1374 but the fix is not contained in any release yet.

Can you retry with with the master branch? You can get a pre compiled beta version here: https://beta.restic.net/v0.7.3-195-g8d37b723/

I'll close this issues as a duplicate of #1351 for now, we can reopen it later if it turns out to be something else.

Please report back!

**Comment by sumnerboy12** [comment]
I have retried using the 0.7.3-195 beta and can confirm pruning now completes without error. Thanks for fixing!!
